mandatory/3-playing-computer.js

```diff
@@ -39,4 +39,16 @@ for (let i = 0; i < 5; ++i) {
     const e = f1(i, a);
     console.log(e);
   }
+  i;
 }
+
+/**
+  * Answers
+  
+  * 1. Ln31 --> b is not defined hence program throws an error
+  * 3. The value of x, a, d and e is printed
+  * 4. f1 is called once
+  * 5. f2 is called once
+  * 6. value is zero
+  * 7. Value is one
+**/
```

Comment:
the loop is running five times,
so, f2 will be called thrice for all even numbers and f1 will be called twice for all odd numbers. remember the loop runs five times. f2 is called for only even numbers and f1 is called for only odd numbers 
at 6. so for the first call of f1, the value of a is 8, remember a is being incremented every time the loop runs but the original value of "a" at the top remains 6 

this question was a bit confusing so I spent much time to figure it out. I am not sure if I am entirely correct. think about my suggestions and let me know if you come up with different ideas. you should try to log a few results to prove yourself. 

otherwise, great job

Reply:
Oh yea I see what you meaning with regards to **how many times is the f1 called I forgot there's a loop there** thanks for pointing that out...

However with regards to this question below:

> What value does the "a" parameter take in the first "f1" call?

On the first call of f1 they are referring to the **a parameter** and the parameter are they not referring to the parameter on the defined function which in this case is occupied by / getting the value of **i** which on the first run is zero because 
```js
 a = a + 1;
```
That above on the first run will result to 7 failing to hit the **if block** jumping to the **else block** where **i** is zero

**What you think?**

Reply:
I see where you are coming from with the 7. There is something I want you to notice though. when the program is run the f2 run first. at this point a is 7. when the loop comes back for the second time, that's when the f1 is run for the first time and a is 8. so it goes like this 

```
step1. run the loop
step2. increment a // now a is 7 because it was initially 6
step3. run f2 ( f2 runs because `i` is 0 and 0 is an even number. at this point the loop goes back without running f1)
step4. go back to running the loop
step5. increment a // now a is 8 
step6. run f1 ( f2 does not run because now `i` is 1 and 1 is an odd number. f1 now runs here for the first time. a will have been incremented twice and a will be 8)

You get it now? Let me know what you think 
```

Reply:
Nah G I hear you 100% this question was confusing and tricky but yea I understand your point you right